src/shared/util/logic.ts

```diff
@@ -334,20 +358,34 @@ const isConditionFulfilled = (
     // TODO: An option that is named "Others: Something..." will also pass this test,
     // even if the field has not been configured to set othersRadioButton=true
     if (conditionValues.indexOf('Others') > -1) {
-      if (field.fieldType === 'radiobutton') {
+      if (field.fieldType === BasicField.Radio) {
         conditionValues.push('radioButtonOthers')
-      } else if (field.fieldType === 'checkbox') {
-        conditionValues.push('checkboxOthers') // Checkbox currently doesn't have logic, but the 'Others' will work in the future if it in implemented
       }
       return (
         conditionValues.indexOf(currentValue) > -1 || // Client-side
         currentValue.startsWith('Others: ')
       ) // Server-side
     }
     return conditionValues.indexOf(currentValue) > -1
-  } else if (condition.state === 'is less than or equal to') {
+  } else if (condition.state === LogicConditionState.AnyOf) {
+    if (
+      field.fieldType === BasicField.Checkbox &&
+      isLogicCheckboxCondition(condition) &&
+      isCheckboxConditionValue(currentValue)
+    ) {
+      condition.value.forEach((obj) => {
+        obj.options = obj.options.sort()
+      })
+      currentValue.options = currentValue.options.sort()
+      return condition.value.some((val) =>
+        isEqual(JSON.stringify(currentValue), JSON.stringify(val)),
```

Comment:
seems a bit sus to just stringify, makes this susceptible to bugs where keys/arrays are not sorted, or there are extra keys etc. can we either use a lodash function to check equality or write a helper function to check equality in the keys that we care about?